cli/flox-activations/src/cli/start_or_attach.rs

```diff
@@ -0,0 +1,81 @@
+use std::path::PathBuf;
+
+use anyhow::Context;
+use clap::Args;
+use time::{Duration, OffsetDateTime};
+
+use crate::activations;
+
+#[derive(Debug, Args)]
+pub struct StartOrAttachArgs {
+    #[arg(help = "The PID of the shell registering interest in the activation.")]
+    #[arg(short, long, value_name = "PID")]
+    pub pid: u32,
+    #[arg(help = "The path to the .flox directory for the environment.")]
+    #[arg(short, long, value_name = "PATH")]
+    pub flox_env: PathBuf,
+    #[arg(help = "The store path of the rendered environment for this activation.")]
+    #[arg(short, long, value_name = "PATH")]
+    pub store_path: PathBuf,
+}
+
+impl StartOrAttachArgs {
+    pub(crate) fn handle(self, cache_dir: PathBuf) -> Result<(), anyhow::Error> {
+        let activations_json_path = activations::activations_json_path(&cache_dir, &self.flox_env)?;
+
+        let (activations, lock) = activations::read_activations_json(&activations_json_path)?;
+        let mut activations = activations.unwrap_or_default();
+
+        let (activation_id, attaching) =
+            match activations.get_or_create_activation_for_store_path(&self.store_path) {
+                activations::NewOrExisting::New(activation) => {
+                    activation.attach_pid(self.pid, None);
+                    (activation.id(), false)
+                },
+                activations::NewOrExisting::Existing(activation) => (activation.id(), true),
+            };
+
+        activations::write_activations_json(&activations, &activations_json_path, lock)?;
+
+        let start_expiration = OffsetDateTime::now_utc() + Duration::seconds(10);
+
+        loop {
+            let (activations, _lock) = activations::read_activations_json(&activations_json_path)?;
+            let Some(mut activations) = activations else {
+                todo!("Who deleted this file?!");
+            };
+
+            let activation = activations
+                .activation_for_id_ref(activation_id)
+                .with_context(|| {
+                    format!(
+                        "No activation with ID {} found for environment {}",
+                        activation_id,
+                        self.flox_env.display()
```

Comment:
nit: you could move the error inside the helper rather than duplicating the error message